Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -975,6 +975,11 @@ class BrowserViewController: UIViewController {
     }
 
     func openURLInNewTab(_ url: URL?, isPrivate: Bool = false, isPrivileged: Bool) {
+        if urlBar.inOverlayMode {
```

Comment:
Is this check necessary, can we just call `urlBar.leaveOverlayMode(didCancel: true)` regardless of our current state?

Reply:
Im not actually sure if it is, it's just how it was setup in previous calling sites. I'll double check.

Reply:
I checked, it seems that `leaveOverlayMode(didCancel:)` makes a delegate call regardless of if it actually left overlay mode (`urlBarDidLeaveOverlayMode`), which would technically incorrect if it didn't leave overlay mode.

Maybe a better solution would be to add the check if its in overlay mode to that method instead like: 

```swift
func leaveOverlayMode(didCancel cancel: Bool = false) {
    if !inOverlayMode { return }
    locationTextField?.resignFirstResponder()
    animateToOverlayState(overlayMode: false, didCancel: cancel)
    delegate?.urlBarDidLeaveOverlayMode(self)
}
```

Reply:
That looks good assuming you are confident the impact won't cause other side effects.

Reply:
I'll have to double-check all the calls to it first, but I think it should be good.